force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -122,20 +139,20 @@ public with sharing class ClassesObjectsMethods {
      * @return A string containing the names of scientists that contain the letter 'g'.
      */
     public static String findTheScientist() {
-        // The list of scientists' names
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
-
-        // The variable to store the concatenated string
-
-        // Loop through the list of scientists' names
-
-            // If the name doesn't contain the letter 'G', skip this iteration
-
-            // Add the name to the result string, followed by a comma
-
-        // Remove the last comma from the result string
-
-        return null; // Replace null with the variable you used to store the result
+        String nameString = '';
+        String name = '';
+        for (Integer i = 0; i < scientistNames.size(); i++) {   
+            name = scientistNames[i];
+            if (name.contains('g') || name.contains('G')) {
+                nameString += name;
```

Comment:
You can add the scientist names to a List, and then use String.join(namesWithG, ',') to combine them into a CSV.